pages/api/dashboard/index.ts

```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
 }
 
 const getUserDashboardData = async function (userId: string): Promise<DashboardData> {
-  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId)).length
+  const buttons = (await paybuttonService.fetchPaybuttonArrayByUserId(userId))
   const addresses = await addressService.fetchAllUserAddresses(userId, true)
   const XECAddresses = addresses.filter((addr) => addr.networkId === XEC_NETWORK_ID)
   const BCHAddresses = addresses.filter((addr) => addr.networkId === BCH_NETWORK_ID)
   const BCHTransactions = Array.prototype.concat.apply([], BCHAddresses.map((addr) => addr.transactions))
   const XECTransactions = Array.prototype.concat.apply([], XECAddresses.map((addr) => addr.transactions))
   const incomingTransactionsInUSD = BCHTransactions.map((t) => {
     t.amount = t.amount.times(DUMMY_BCH_PRICE)
+    t.network = 'BCH'
+    t.button = buttons.find(button => button.addresses.find(add => add.address.id === t.addressId))
```

Comment:
What if you passed all the buttons? Then in the front end there could be link to all buttons that have the address of the transaction.

You would need only to pass something like
```
t.buttons = buttons.filter(button => button.addresses.some(add => add.address.id === t.addressId)).map((button) => {
  return {
    name: button.name,
    id: button.id
  }
})
```